readme.md

```diff
@@ -191,24 +194,88 @@ Print is the only language defined i/o function. It takes in a single variable a
 The following are valid print statements:
 ```
 print ( "hello" );
-print ( a );
+print ( 42 );
+print ( a );  # Only valid if the variable "a" is defined.
 ```
 
 ### Loops
 A loop will itterate between a range of numbers.
+
 Example loops:
 ```
-loop ( a in range, ( 1, 100 ) ) {
+loop ( a in range( 1, 100 ) ) {
 
 	print ( a );
 
 }
 ```
+### Type casting
+Most data types can be converted into any other data type.
+```
+# Convert to int.
+# Can be done with all data types.
+int(x)
+
+# Convert to float.
+# Can be done with all data types.
+float(x)
+
+# Convert to string.
+# Can be done with all data types.
+string(x)
+
+# Convert to bool.
+# Can be done with all data types.
+bool()
+
+# Convert to none.
+# Can be done with all data types.
+none(x)
+```
+
+### NOP
+Incase you need something to do nothing, you can use the `pass` keyword to do nothing.
+
+Example:
+```
+if 2 > 1 {
+
+ print ( "hello" );
+
+} 
+else { 
```

Comment:
I think this could be better demonstrated via a function that’s passing rather than an else

Reply:
I am a bit confused about what you mean, could you elaborate?

Reply:
Well for instance let’s say you have a function:
```
hello( variable ) {
    variable = variable + 1
    pass
}
``` 
It doesn’t return the new variable, it returns nothing

Reply:
Another option would be to allow `return none`

Reply:
It isn't possible currently?

Reply:
Well not if you have `none( thing ) ` be possible. See discussion on none as a type

Reply:
Do functions have to return something in Sack?

Reply:
By default no

Reply:
Then why use `pass` in a function that already has a statement?

Reply:
Same reason why some people put return; in void functions, for completeness' sake
-Alvarito

Reply:
bruh
\- Alvarito

Reply:
![image](https://user-images.githubusercontent.com/47306082/151880991-cce353f9-975c-4ad9-9288-07a1b24a774c.png)